routing/turns_tts_text.cpp

```diff
@@ -47,10 +48,69 @@ void GetTtsText::ForTestingSetLocaleWithJson(std::string const & jsonBuffer, std
   m_getCurLang = platform::ForTestingGetTextByIdFactory(jsonBuffer, locale);
 }
 
+/**
+ * @brief Modified version of GetFullRoadName in routing_session.cpp.
+ * For next street returns "ref; name" .
+ * For highway exits (or main roads with exit info) returns "junction:ref; target:ref; target".
+ * If no |target| - it will be replaced by |name| of next street.
+ * If no |target:ref| - it will be replaced by |ref| of next road.
+ * So if link has no info at all, "ref; name" of next will be returned (as for next street).
+ *
+ * @arg RouteSegment::RoadNameInfo & road - structured info input about the next street
+ * @arg std::string & name - semicolon-delimited string output for the TTS engine
+ */
+void FormatFullRoadName(RouteSegment::RoadNameInfo & road, std::string & name)
+{
+  name.clear();
+
+  if (auto const & sh = ftypes::GetRoadShields(road.m_ref); !sh.empty())
+    road.m_ref = sh[0].m_name;
+  if (auto const & sh = ftypes::GetRoadShields(road.m_destination_ref); !sh.empty())
+    road.m_destination_ref = sh[0].m_name;
+
+  std::vector<std::string> outArr;
+
+  if (road.HasExitInfo())
+  {
+    if (!road.m_junction_ref.empty())
+      outArr.emplace_back(road.m_junction_ref);
+
+    if (!road.m_destination_ref.empty())
+      outArr.emplace_back(road.m_destination_ref);
+
+    if (!road.m_destination.empty())
+      outArr.emplace_back(road.m_destination);
+    else if (!road.m_name.empty())
+      outArr.emplace_back(road.m_name);
+  }
+  else
+  {
+    if (!road.m_ref.empty())
+      outArr.emplace_back(road.m_ref);
+    if (!road.m_name.empty())
+      outArr.emplace_back(road.m_name);
+  }
+
+  if (outArr.size() > 0)
+  {
+    // append strings with delimiter and no trailing
+    for (size_t i = 0; i < outArr.size() - 1; i++)
+      name.append(outArr[i] + "; ");
+    name.append(outArr[outArr.size() - 1]);
+  }
```

Comment:
```suggestion
  name = strings::JoinStrings(outArr.begin(), outArr.end(), "; ");
```

Remove `name.clear()` call in the beginning, remove name as a parameter, and remove it instead as a result of the function instead of `void`.